core/src/main/java/com/linecorp/armeria/client/HttpSessionHandler.java

```diff
@@ -562,4 +579,44 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws E
             // Netty will close the connection automatically on an IOException.
         }
     }
+
+    private void applyOutlierDetection(ClientRequestContext ctx) {
+        if (outlierDetectingRule == null || !isAcquirable()) {
+            return;
+        }
+        assert outlierDetector != null;
+        if (outlierDetector.isOutlier()) {
+            return;
+        }
+
+        ctx.log().whenComplete().thenAccept(log -> {
+            final OutlierDetectionDecision decision =
+                    outlierDetectingRule.decide(ctx, log.responseHeaders(), log.responseCause());
```

Comment:
Question) Is there no need need to at least have some basic protection against exceptions since `outlierDetectingRule` is an arbitrary function?

Reply:
Since there is no additional chained operation, it seems safe to just raise an error.

Reply:
I see, I guess it's fine as long as users are signalled somehow that an exception has been thrown

Reply:
Fixed to warn an exception raised.